source/mechanism.md

```diff
@@ -115,14 +115,46 @@ The primary aim with this deployment was to provide the [dxDAO](https://dutchx.r
 
 ### Some more details to the auctioneer powers:
 #### De- & whitelisting of tokens that generate Magnolia
-TODO
+Whitelisted tokens - if traded in a whitelisted pair - generate Magnolia, which can be used to reduce liquditiy contribution.
+Read above about [Magnolia](https://dutchx.readthedocs.io/en/latest/mechanism.html#magnolia), [Whitelist](https://dutchx.readthedocs.io/en/latest/mechanism.html#whitelist) and [Liquidity contribution](https://dutchx.readthedocs.io/en/latest/mechanism.html#liquidity-contribution).<br/>
+- Whitelisting (and de-whitelisting) can happen by token individually or by entire lists
+- It is not necessary that the token is either listed for trading or actively traded
+- The auctioneer calls the function *updateApprovalOfToken* in the [Token Whitelist contract](https://github.com/gnosis/dx-contracts/blob/master/contracts/base/TokenWhitelist.sol#L28)
```

Comment:
> in the Token Whitelist contract

and all other `in the * contract`  crate an impression that they are all different contracts when in reality its all DutchExchange that inherits from them.
Better would be

> calls the function *updateApprovalOfToken* on the DutchExchangeProxy contract which is implemented in the [Token Whitelist contract](link)

or

> ... (see implementation in [Contract](link))

Reply:
or like your
> the function * of the [Implementation Contract] (via the [Proxy])

below